httpcore5/src/test/java/org/apache/hc/core5/net/TestURIBuilder.java

```diff
@@ -355,8 +440,8 @@ public void assertSetParameters(final Charset charset) throws Exception {
     }
 
     public void assertBuild(final Charset charset, final URI uri) throws Exception {
-        final String encodedData1 = URLEncodedUtils.encodeFormFields("\"1\u00aa position\"", charset);
-        final String encodedData2 = URLEncodedUtils.encodeFormFields("Jos\u00e9 Abra\u00e3o", charset);
+        final String encodedData1 = PercentCodec.encode("\"1\u00aa position\"", charset);
```

Comment:
Swapping form to URI encoding looks wrong, isn't it?

Reply:
@michael-o Why? The entire `URLEncodedUtils` got completely messed by mixing up `application/x-www-form-urlencoded` or URI coding logic hence its deprecation.

Reply:
I understand that, but when I try to encode `hello there` form and URI encoding will give me different results.